source/kann.js

```diff
@@ -0,0 +1,322 @@
+import * as flatbuffers from './flatbuffers.js';
+
+const kann = {};
+
+kann.ModelFactory = class {
+
+    // Check whether a KaNN file is processed and initialize the container
+    match(context) {
+        // Context is an object providing necessary information to interact with the processed file
+        const container = kann.Container.open(context); // attempt to get a container from the context
```

Comment:
This can mostly be removed and simplified as FlatBuffers detection in archives is mostly automatic. See `tflite.js` for an example. Are these files always `.tar.gz` without a file extension or mostly regular FlatBuffers files like TensorFlow Lite?

Reply:
They are always `.tar.gz` containing an inner FlatBuffers `.kann`

Reply:
Is there a guideline that such containers should always be named `.kann.tar.gz` or `.tar.gz`. Naming both the inner FlatBuffers file and the container `.kann` is confusing?

Reply:
The container isn't name `.kann` anymore, sorry for the confusion, but what should this guideline be ?

Reply:
done